### DISCUSSION TITLE
Laravel artisan serve

### DISCUSSION THREAD

**Discussion by mark-win** [NoiseByNorthwest/php-spx#246] [Show and tell]
I got this to work in a local Laravel dev environment and wanted to share how I did it.

Running on Ubuntu, I created the following `/etc/php/8.3/mods-available/spx.ini`
```
extension=spx.so
spx.http_enabled=1
spx.http_key="dev"
spx.http_ip_whitelist="127.0.0.1"
spx.http_profiling_auto_start=0
spx.http_profiling_enabled=1
```
Then I created a link, to enable it in cli:
```
sudo ln -s /etc/php/8.3/mods-available/spx.ini /etc/php/8.3/cli/conf.d/30-spx.ini
```

As `artisan serve` is long running, `spx.profiling_auto_start` is disabled in the ini and needs to be handled manually by code. So I created a middleware, that triggers in local environment.
```
    public function handle(Request $request, Closure $next): Response
    {
        $local = config('app.env') === 'local';
        if($local && function_exists('spx_profiler_start')) {
            spx_profiler_start();
        }
        $return = $next($request);
        if ($local && function_exists('spx_profiler_stop')) {
            spx_profiler_stop();
        }

        return $return;
    }
```
This is then registered in `App\Http\Kernel::$middleware` as the first entry.
Depending on your needs, it might be wise to check for a custom env var, or even make it a group middleware to have more control over the profiled routes.

`php artisan serve` can be called as usual and the web panel can be reached as expected: `http://localhost:8000/?SPX_KEY=dev&SPX_UI_URI=/`.

@NoiseByNorthwest thank you for the great tool!

![grafik](https://github.com/NoiseByNorthwest/php-spx/assets/17962085/414bb986-afd1-4a88-a970-15f2e2a88a31)

**Comment by NoiseByNorthwest** [comment]
This is the way to do it with via a middleware, do not forget the `try ... finally` block around the call to `$next` https://github.com/NoiseByNorthwest/php-spx?tab=readme-ov-file#handle-long-living--daemon-processes

**Reply by mark-win** [reply]
You are right. Thank you for the hint!
